lib/srv/discovery/fetchers/aws-sync/s3.go

```diff
@@ -201,16 +212,12 @@ type s3Details struct {
 	tags         *s3.GetBucketTaggingOutput
 }
 
-func (a *awsFetcher) getS3BucketDetails(ctx context.Context, bucket *s3.Bucket, bucketRegion string) (s3Details, failedRequests, []error) {
+func (a *awsFetcher) getS3BucketDetails(ctx context.Context, bucket s3types.Bucket, bucketRegion string) (s3Details, failedRequests, []error) {
 	var failedReqs failedRequests
 	var errs []error
 	var details s3Details
 
-	s3Client, err := a.CloudClients.GetAWSS3Client(
-		ctx,
-		bucketRegion,
-		a.getAWSOptions()...,
-	)
+	awsCfg, err := a.AWSConfigProvider.GetConfig(ctx, bucketRegion, a.getAWSV2Options()...)
```

Comment:
nit: should we update the error message below? `failed to create s3 client for bucket...`